src/index.ts

```diff
@@ -93,14 +97,37 @@ const config = Config.fromFile("./config.toml");
   const comitNode = new ComitNode(config);
   const datastore = new DefaultFieldDataSource(ledgerExecutorParams);
   const ledgerExecutor = new LedgerExecutor(ledgerExecutorParams);
-  const actionSelector = new ActionSelector(config, tradeEvaluationService);
+  const actionSelector = new ActionSelector(config, tradeService);
 
   const actionExecutor = new ActionExecutor(
     comitNode,
     datastore,
     ledgerExecutor
   );
 
+  const comitNodeMetaData = await comitNode.getMetaData();
+  if (!comitNodeMetaData) {
+    throw new Error("Could not retrieve meta-data from comit-node");
+  }
+  const peerId = comitNodeMetaData.id;
+
+  if (!config.bitcoinConfig) {
+    throw new Error("Bitcoin not initialized correctly");
+  }
+  if (!ethereumParams.ethereumWallet) {
+    throw new Error("Ethereum not initialized correctly");
+  }
+
+  api.get(
+    "/trades/publish",
```

Comment:
```suggestion
    "/trades",
```
If it's available on the HTTP API then it's intrinsically published.